lib/rucio/core/replica.py

```diff
@@ -799,6 +803,37 @@ def _list_replicas_for_files(file_clause, state_clause, files, rse_clause, updat
             {'scope': scope, 'name': name} in files and files.remove({'scope': scope, 'name': name})
 
 
+def get_multi_cache_prefix(cache_site, filename):
+
+    # print('Looking up prefix for cache:', cache_site, 'and filename:', filename)
+    x_caches = read_from_cache('CacheSites', expiration_time=60)
+    if not x_caches:
+        # print('reloading xcache sites and ranges.')
+        vp_endpoint = 'https://vps.cern.ch/serverRanges'
+        response = requests.get(vp_endpoint, verify=False)
+        if response:
+            # print('Success! Got XCache ranges.')
+            x_caches = response.json()
+            # print(x_caches)
+            write_to_cache('CacheSites', x_caches)
+        else:
+            print('Could not reload from vps.')
```

Comment:
use a logger instead of print, see https://github.com/rucio/rucio/commit/e433c472abcc54536d111371466f6af934c3cede#diff-1085ef0a38068abd1b394d8a69b43fa31b34a3a3eb1686fc3727f3638853c7cdR558 on how this is done in core now.

Reply:
I will try.